iphone/Classes/MediaModule.m

```diff
@@ -1755,6 +1775,150 @@ - (void)handleTrimmedVideo:(NSURL *)theURL withDictionary:(NSDictionary *)dictio
+                             } else {
+                               [imageArray addObject:@{ @"error" : error.description, @"code" : @(error.code), @"success" : @(NO), @"mediaType" : (NSString *)kUTTypeImage }];
+                               DebugLog(@"[ERROR] Failed to load image- %@ .", error.description);
+                             }
+                             dispatch_group_leave(group);
+                           }];
+    } else if ([result.itemProvider hasItemConformingToTypeIdentifier:UTTypeMovie.identifier]) {
+      [result.itemProvider loadItemForTypeIdentifier:UTTypeMovie.identifier
+                                             options:nil
+                                   completionHandler:^(__kindof id<NSSecureCoding> _Nullable item, NSError *_Null_unspecified error) {
+                                     // Bug from Apple : https://developer.apple.com/forums/thread/652695
+                                     if (!error) {
+                                       if (!videoArray) {
+                                         videoArray = [[NSMutableArray alloc] init];
+                                       }
+                                       NSURL *url = (NSURL *)item;
+                                       TiBlob *media = [[[TiBlob alloc] initWithFile:[url path]] autorelease];
+                                       if ([media mimeType] == nil) {
+                                         [media setMimeType:@"video/mpeg" type:TiBlobTypeFile];
+                                       }
+                                       [videoArray addObject:@{ @"media" : media, @"mediaType" : (NSString *)kUTTypeMovie, @"success" : @(YES), @"code" : @(0) }];
+                                     } else {
+                                       [videoArray addObject:@{ @"error" : error.description, @"code" : @(error.code), @"success" : @(NO), @"mediaType" : (NSString *)kUTTypeMovie }];
+                                       DebugLog(@"[ERROR] Failed to load video- %@ .", error.description);
+                                     }
+                                     dispatch_group_leave(group);
+                                   }];
+    } else {
+      dispatch_group_leave(group);
+      NSLog(@"Unsupported media type");
+    }
+  }
+
+  dispatch_group_notify(group, dispatch_get_global_queue(QOS_CLASS_DEFAULT, DISPATCH_QUEUE_PRIORITY_DEFAULT), ^{
+    // Perform completition block
+    NSMutableDictionary *dictionary = [TiUtils dictionaryWithCode:0 message:nil];
+    if (livePhotoArray != nil) {
+      [dictionary setObject:livePhotoArray forKey:@"livePhotos"];
+    }
+    if (imageArray != nil) {
+      [dictionary setObject:imageArray forKey:@"images"];
+    }
+    if (videoArray != nil) {
+      [dictionary setObject:videoArray forKey:@"videos"];
+    }
+    [self sendPickerSuccess:dictionary];
+  });
+
+  if (autoHidePicker) {
+    [self closeModalPicker:picker];
+  }
+}
```

Comment:
You are on a background thread here. Maybe validate if `closeModalPicker` and `sendPickerSuccess` really are performed on the main thread. In any case, there doesn't seem to be much processing here, so is a global queue / background queue necessary?

Reply:
There is no special handling in this delegate method for `closeModalPicker` and  `sendPickerSuccess`. It is same as with old picker. In `dispatch_group_notify` queue is mandatory to pass.